src/btree/bt_discard.c

```diff
@@ -240,6 +240,28 @@ __free_page_modify(WT_SESSION_IMPL *session, WT_PAGE *page)
     __wt_free(session, page->modify);
 }
 
+/*
+ * __ref_addr_safe_free --
+ *     Any thread that is reviewing the address in a WT_REF, must also be holding a split generation
+ *     to ensure that the page index they are using remains valid. Use that same split generation to
+ *     ensure that the address inside the WT_REF remains valid while it is being reviewed.
+ */
+static void
+__ref_addr_safe_free(WT_SESSION_IMPL *session, void *ref_addr)
+{
+    WT_DECL_RET;
+    uint64_t split_gen;
+
+    split_gen = __wt_gen(session, WT_GEN_SPLIT);
+    WT_TRET(__wt_stash_add(
+      session, WT_GEN_SPLIT, split_gen, ((WT_ADDR *)ref_addr)->addr, ((WT_ADDR *)ref_addr)->size));
+    WT_TRET(__wt_stash_add(session, WT_GEN_SPLIT, split_gen, ref_addr, sizeof(WT_ADDR)));
+    __wt_gen_next(session, WT_GEN_SPLIT, NULL);
```

Comment:
Can we have a comment here (or on the `wt_gen` call) explaining that no splits are taking place, but we're using the generation since we know readers must hold one?

Reply:
I tried to add some comments to explain why WT_GEN_SPLIT is used, please check and let me know.

Reply:
Looks good, thanks!